### DISCUSSION TITLE
Guidance on how to create a exportable package with Prisma.

### DISCUSSION THREAD

**Discussion by JoshuaPuello** [prisma/prisma#23055] [Q&A] [answered]
### Question

I have a NestJs project with yarn and Prisma already configured. My intention with this project is to publish it to the gitlab package registry so I can import and use Prisma models in other packages. 

This is my current configuration: 

![image](https://github.com/prisma/prisma/assets/33018356/4ae72757-fab7-489f-8ba4-93cd87f5e6cc)

<details>

<summary>schema.prisma:</summary>

```
generator client {
  provider = "prisma-client-js"
  output   = "../dist/generated/client"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
...
}
```

</details>

<details>

<summary>index.ts:</summary>

```
export { PrismaClient } from '@prisma/client'
```

</details>

<details>

<summary>package.json:</summary>

```
{
  "name": "@company/xxx-orm",
  "description": "Shared orm models and configuration.",
  "version": "0.0.20",
  "license": "UNLICENSED",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "scripts": {
    "format": "prettier --write .",
    "migrate:run": "yarn prisma migrate dev",
    "migrate:create": "yarn prisma migrate dev --create-only",
    "build": "tsc && yarn prisma generate",
    "prepublishOnly": "yarn build"
  },
  "packageManager": "yarn@3.8.0",
  "dependencies": {
    "@prisma/client": "5.8.1"
  },
  "devDependencies": {
    "prettier": "^3.2.2",
    "prisma": "^5.8.1",
    "typescript": "^4.2.3"
  },
  "publishConfig": {
    "registry": "https://gitlab.xxx.org/api/v4/projects/13/packages/npm/"
  }
}
```

</details>

This is successfully generating the `dist/generated/client` folder. 

![image](https://github.com/prisma/prisma/assets/33018356/db4dfb49-fcde-4f1d-a436-5e873e09b9fd)

### Problem

I'm publishing the package and importing it in a different project, but when I attempt to do a `new PrismaClient()` I get the following error: 

```
[Nest] 91163  - 02/09/2024, 10:52:42 AM   ERROR [ExceptionsHandler] @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
```

This is the generic error that is in `@prisma/client`, I thought of maybe my `index.ts` should point to the `dist/generated/client` instead since that's the place where the actual files are being generated, but I got: 

```
Error: Cannot find module 'dist/generated/client'
```

I hope someone can give me some guidance on what should be changed or added here. Thank you. 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: macOS
- Database: PostgreSQL
- Node.js version: v21.5.0
- Yarn version: 3.8.0

**Comment by ludralph** [comment, marked as answer]
Hi @JoshuaPuello 👋 

Thank you for raising this question.

Can you please try generating the types in the default location (of node_modules) by removing this output from the schema file
```
output   = "../dist/generated/client"
```

and export it like

```
export { PrismaClient } from '@prisma/client'
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by GHkrishna** [comment]
Hello @JoshuaPuello @ludralph 

Do you mind sharing, how did we export all the types from prisma (schema)?
